content/server/ha.md

```diff
@@ -0,0 +1,214 @@
+
+- makes the assumption we are running postgres db on the same host as the drone servers.
+- makes the assumption we are running redis db on the same host as the drone servers. This is the trigger for Drone to run in HA mode.
+
+For drone server 1
+
+```bash
+DRONE_SERVER_HOST=localhost:8081
+DRONE_SERVER_PORT=:8081
+DRONE_SERVER_PROTO=http
+DRONE_DATABASE_DRIVER=postgres
+DRONE_DATABASE_DATASOURCE=postgres://postgres:postgres@localhost:5432/drone?sslmode=disable
+DRONE_REDIS_CONNECTION=redis://localhost:6379
+```
+
+For drone server 2
+
+```bash
+DRONE_SERVER_HOST=localhost:8082
+DRONE_SERVER_PORT=:8082
+DRONE_SERVER_PROTO=http
+DRONE_DATABASE_DRIVER=postgres
+DRONE_DATABASE_DATASOURCE=postgres://postgres:postgres@localhost:5432/drone?sslmode=disable
+DRONE_REDIS_CONNECTION=redis://localhost:6379
+```
+
+## HAProxy
+
+HA Proxy can work on layer 4 or layer 7 (OSI model) mode. Both modes can be used. Example `haproxy.cfg` files follow. HA proxy can be start with the following `haproxy -f haproxy.cfg` command.
+
+Now, the Drone frontend app can be opened at http://localhost:8080. 
+
+### Layer 4 mode
+
+```bash
+frontend drone
+    bind *:8080
+    default_backend drone_servers
+    timeout client 120s
+
+backend drone_servers
+    balance roundrobin
+    timeout connect 5s
+    timeout server 120s
+    server server1 localhost:8081
+    server server2 localhost:8082
+```
+
+### Layer 7 mode
+
+Note that in layer 7 mode HTTP responses will have an additional header `X-App-Server` that tells which server processed the request.
+
+```bash
+defaults
+    mode http
+    timeout connect 5s
+    timeout client 120s
+    timeout server 120s
+    maxconn 128
+
+frontend drone
+    bind *:8080
+    default_backend drone_servers
+
+backend drone_servers
+    balance roundrobin
+    server server1 localhost:8081
+    server server2 localhost:8082
+    http-response add-header X-App-Server %b/%s
+
+```
+
+# Docker compose example
+
+***Note: This is for demonstration purposes only, and not for production.***
+
+Running all the containers in a single machine does not make sense, this example is provided as an example of a HA deployment.
+
+You will need to create these two files in a directory and subsitiute in correct values to the docker-compose.yml file.
+
+Start the stack via `docker-compose up`.
+
+## `haproxy.cfg` file
+
+```bash
+defaults
+    mode http
+    timeout connect 5s
+    timeout client 120s
+    timeout server 120s
+    maxconn 128
+
+frontend drone
+    bind *:8080
+    default_backend drone_servers
+
+backend drone_servers
+    balance roundrobin
+    server server1 dronea:8081
+    server server2 droneb:8082
+    http-response add-header X-App-Server %b/%s
+```
+
+## `docker-compose.yml` file
+
+```yaml
+version: "3.8"
+services:
+    dronea:
+        image: drone/drone:latest
+        depends_on: 
+            db:
+                condition: service_healthy
+        environment:
+        - DRONE_SERVER_HOST=localhost:8081
```

Comment:
should be `DRONE_SERVER_HOST=localhost:8080` which is the public facing (load balancer) address